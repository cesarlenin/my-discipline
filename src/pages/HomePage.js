import React, { Component } from 'react'
import { Link} from 'react-router-dom'
import HabitList from '../components/HabitList';
import UserContext from '../components/UserContext';
import HabitsApiService from '../services/my-discipline-api-service'

export class HomePage extends Component {
  static contextType = UserContext;

  componentDidMount() {
    if(this.context.authToken){
      this.context.clearError()
      HabitsApiService.getHabits()
      .then(data=>this.context.setHabits(data))
      .catch(error=>this.context.setError(error))
  
      this.context.clearError()
      HabitsApiService.getActions()
      .then(data=>this.context.setActions(data))
      .catch(error=>this.context.setError(error))
    }
  }


  render() {
    return (
    <main>
      <header>
        <h1>Habit Collection</h1>
      </header>

      <section className= "aButton">
      <Link to='/createHabit'>
      create habit
      </Link>
      </section>

      <HabitList/>
    </main>
    )
  }
}

export default HomePage