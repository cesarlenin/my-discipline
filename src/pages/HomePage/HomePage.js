import React, { Component } from 'react'
import { Link} from 'react-router-dom'
import HabitList from '../../components/HabitList/HabitList';
import UserContext from '../../components/UserContext/UserContext';
import HabitsApiService from '../../services/my-discipline-api-service'
import './HomePage.css'

export class HomePage extends Component {
  static contextType = UserContext;

  componentDidMount() {
    if(this.context.authToken){
      this.context.clearError()
      HabitsApiService.getHabits()
      .then(data=>this.context.setHabits(data))
      .catch(error=>this.context.setError(error))
  
      this.context.clearError()
      HabitsApiService.getActions()
      .then(data=>this.context.setActions(data))
      .catch(error=>this.context.setError(error))
    }
  }


  render() {
    console.log(this.context.habits)
    return (
    <section>
      <header>
        <h1>My Habits</h1>
      </header>

      <section className= "create">
      <Link to='/createHabit'>
      Create Habit
      </Link>
      </section>

      <HabitList {...this.props}/>
    </section>
    )
  }
}

export default HomePage