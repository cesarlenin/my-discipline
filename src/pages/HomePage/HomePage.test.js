import React from 'react';
import ReactDOM from 'react-dom';
import HomePage from './HomePage';
import store from '../../dummy-store';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

describe('<HomePage />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
  <HomePage
  habit={store.habits}
  />
  </BrowserRouter>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders this UI as expected', () => {
  const tree = renderer.create(
<BrowserRouter>
  <HomePage
  habits={store.habits}
  />
  </BrowserRouter>).toJSON();
  expect(tree).toMatchSnapshot();
});

});