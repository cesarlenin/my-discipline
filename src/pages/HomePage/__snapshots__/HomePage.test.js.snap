// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders this UI as expected 1`] = `
<section>
  <header>
    <h1>
      My Habits
    </h1>
  </header>
  <section
    className="create"
  >
    <a
      href="/createHabit"
      onClick={[Function]}
    >
      Create Habit
    </a>
  </section>
  <div />
</section>
`;
